src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,50 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.UserNotOkException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_CHARS = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        try {
+            checkingUser(user);
+            storageDao.add(user);
+            return storageDao.get(user.getLogin());
+        } catch (UserNotOkException e) {
+            throw new RuntimeException(e);
+        }
```

Comment:
The `register` method should not catch the `UserNotOkException` only to rethrow it as a `RuntimeException`. This is unnecessary wrapping of exceptions and can obscure the original cause of the error. It's better to let the `UserNotOkException` be thrown directly, as it provides more specific information about what went wrong.